### PR TITLE
Add utilities for opening files, including a data directory inside chemper installed with pip, and condensed smirks output

### PR DESCRIPTION
**Pull request by bannanc**
In this pull request I have a jupyter notebook to show some initial attempts at how to shrink the complex SMIRKS patterns currently created by `ClusterGraph`. 

This led to a few improvements and utility functions in the rest of the code:
* read in molecules from mol2 files
    - I'm still deciding how much of this we will need to support in the long run as the goal was to allow people to use their own cheminformatics package while interfacing with my tools. However, having functions to get molecules out of mol2 files makes writing examples easier. 
    - The mol2 function for rdkit is a bit convoluted as it seems the current implementation for rdkit expects only one molecule/file (I don't know why this is the case or if it is true with other file types). I found a solution on sourceforge from 2011, the file support doesn't seem to have changed since then. 
* added `__str__` functions to the molecule classes
* created a condensed SMIRKS format for atoms in `ClusterGraph` which moves decorators that are common to the end of the atom string with an AND "`;`" 
    - For example imagine you are looking at the C-H bond for carbon 1 or 2 in 1-propene: instead of something like `[#6AH2X3x0r0+0,#6AH1X3x0r0+0:1]-;!@[#1AH0X1x0r0+0]` you would get `[#6H2,#6H1;AX3x0r0+0:1]-;!@[#1AH0X1x0r0+0]`. 
    - I'm imagining my next step will be to either systematically or inside some sampling algorithm we will remove decorators until they are unnecessary. 

Below are the things I'm still working on 
- [x] I'm working on making a new example that will show the extent of the `ClusterGraph` technology as it stands.  
    - currently, it is in the folder, but just a place holder I'm playing around and testing things in right now, you shouldn't look at it. 
- [x] I need to add tests for the new `SMIRKS`
    - I honestly haven't tested this at all yet, so I need to make sure it is really working as expected.

### CODE REVIEW

**Comment by bannanc** [comment]
As a side note - I recently noticed that you can now label PRs the same as issues, so I added a WIP label and am using that instead of [WIP] in the title.

**Comment by bannanc** [comment]
OK, this PR is getting giant, but I'm still not done debugging or adding tests for the new code. Some updated:

- I discussed included this in an updated to David, but I decided to try more stocastic approach for removing unnecessary decorators, basically make the Giant Cluster SMIRKS patterns and then remove decorators from them for a set number of iterations, only keep these removals if the automatically generated set matches the specified clustering.

- This lead to needing some tools which are in [openforcefield](), primarily `ChemicalEnvironments` which allows for these changes to SMIRKS patterns. I've only updated that code to be chemper compatible. Once I have this working and `openforcefield` supports both OE and RDK I'll just make it a dependency. 
    - as a side note here, I assigned myself an issue on `ChemicalEnvironments` a long time ago to make it more pythonic. Specifically, I'd like to make the names more pep8 compatible and use the more pythonic `@property` and `@setter` functions instead of the Getters and Setters.

**Comment by bannanc** [comment]
As noted in issue #33 the OE license is expired. I'll make a separate PR to add the new one tomorrow, but both tests pass locally and the RDK tests pass here.

**Comment by bannanc** [comment]
OK, this PR has snow balled A LOT, but I think it is read to go. Included here are a lot of utility functions and the introduction of the [Reducer class](https://github.com/MobleyLab/chemper/blob/smirks_squashing/chemper/optimize_smirks/reducer.py). I still have some work to do on this class, but I think we need to get this infrastructure merged. The new notebook [Generating Reasonable SMIRKS](https://github.com/MobleyLab/chemper/blob/smirks_squashing/examples/Generating%20Reasonable%20SMIRKS.ipynb) is the best demonstration of how the `Reducer` class works currently and also includes documentation of what needs to be explored next. 

I'll put in a PR with a new OE license in a few minutes, then this will be ready to go.

**Comment by bannanc** [comment]
@vtlim is going to help me with at least a little code review before I merge this.

**Comment by vtlim** [comment]
Comments on the ipynb of Generating reasonable SMIRKS.

**Description**
- [ ] What is the assigned clustering for? In other words, if you have grouped bonds, are you trying to find subgroups within that group?
- [ ] Typo of "indice"
- [ ] Item 3 in list is kind of confusing. You might be able to explain this a bit in the text before the steps.

**Step 1**
- no comments

**Step 2**
- [ ] the order in which you list the bonds is not consistent with the description
- [ ] This should be two sentences: "This is done with the utility function... the output comes in"
- [ ] What's the difference between the first label and the smirks label? Are they the same? If not, could you illustrate this?
- [ ] Not sure what is meant by: "There is a list of each dictionaries for each molecule" -- Is it accurate to say: "Each molecule has a list of dictionaries for the SMIRKS-labeled atoms in the molecule."
- [ ] The example description is confusing since it seems like you have a single example but you specify two molecules. You could word it something like, "Two examples are discussed for aromatic bonds. The first molecule in the given SMILES strings is pentane(?), which has zero aromatic bonds, so the first list is empty. The second molecule is benzene, so the second element of `cluster_list[0]` is a non-empty dictionary specifying the ..."
- [ ] The ordering of the `cluster_list` is very non-intuitive here since it looks like `cluster_list[0]` should be double like the way you specify in `smirks_labels` (which is also a different order than you wrote in the description).

**Step 3**
- [ ] Similarly to how the step 3 description was confusing above, you could reiterate the goal  of this step as the first sentence for this section (instead of only saying what is done)
- [ ] typo: "using ALL possible decorator" (should be decorators)
- [ ] typo: "arguement"
- [ ] 3b/3c: can you make these subtitles more descriptive?

**Step 4**

- [ ] our search for what?
- [ ] 4.1: what is the significance of this?
- [ ] 4.2: should be a question. Make sure to specify that it's the removal patterns for SMIRKS decorators in the question.
- [ ] typo: "gaurentee"

File: chemper/optimize_smirks/reducer.py [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+    Generates complex SMIRKS for a given cluster of substructures 
+    and then reduces the decorators in those smirks
+    """
+    def __init__(self, molecules, cluster_list,
+                 layers=2, verbose=True):
+        """
+        Parameters
+        ----------
+        mols: list of chemper Mols
+        cluster_list: list of labels and smirks_atom_lists
```
Comment (vtlim) [inline chemper/optimize_smirks/reducer.py]: The description for `cluster_list` is very confusing

**Comment by bannanc** [comment]
Thanks @vtlim 

For the record on this PR:
* I'm going to make the changes Vickie suggested for the jupyter notebook in a new commit tonight then merge. 
* The cluster list is a mess because of the expected input for `ClusterGraph` I plan on fixing both to be more user intuitive, but in its own PR since it will require updating those examples as well.

**Comment by bannanc** [comment]
I merged this, there are TODOs in the code and I have some experiments I want to run with the best "moves" in the `SMIRKSifier`, but those can happen in smaller steps. I did incorporate all of Vickie's feedback in my notebook.

**Pull request merged**